src/toil/batchSystems/abstractBatchSystem.py

```diff
@@ -168,10 +168,14 @@ def workerCleanup(workerCleanupInfo):
         relevant information for cleaning up the worker.
         """
         assert workerCleanupInfo.__class__.__name__ == 'WorkerCleanupInfo'
-        workflowDir = getToilWorkflowDir(workerCleanupInfo.workflowID, workerCleanupInfo.workDir)
+        workflowDir = Toil.getWorkflowDir(workerCleanupInfo.workflowID, workerCleanupInfo.workDir)
         dirIsEmpty = os.listdir(workflowDir) == []
         cleanWorkDir = workerCleanupInfo.cleanWorkDir
-        if cleanWorkDir == 'always' or ((cleanWorkDir == 'onSuccess' or cleanWorkDir == 'onError') and dirIsEmpty):
+
+        cleanAlways = cleanWorkDir == 'always'
+        cleanOnSuccessOrError = ((cleanWorkDir == 'onSuccess' or cleanWorkDir == 'onError') and dirIsEmpty)
+
+        if cleanAlways or cleanOnSuccessOrError:
```

Comment:
This I find convoluted. How about

```
 if cleanWorkDir == 'always' or dirIsEmpty and cleanWorkDir in ('onSuccess', 'onError'):
```

Generally speaking, I believe that one should exploit the operator precedence and associativity rules except for exotic operators. Everyone should know that `and` binds stronger than `or`, just like any 4th grader knows that * goes before +. In those cases, I would omit the redundant parentheses.